azurerm/resource_arm_dev_test_lab_schedule_test.go

```diff
@@ -0,0 +1,561 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+)
+
+func TestAccAzureRMDevTestLabSchedule_autoShutdownPropertyRulesCheck(t *testing.T) {
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	missingRecurrenceConfig := testAccAzureRMDevTestLabSchedule_autoShutdownMissingRecurrence(ri, location)
+	missingDailyRecurrenceConfig := testAccAzureRMDevTestLabSchedule_autoShutdownMissingDailyRecurrence(ri, location)
+	incorrectTaskNameConfig := testAccAzureRMDevTestLabSchedule_autoShutdownIncorrectTaskName(ri, location)
+	taskNameTaskTypeMismatchConfig := testAccAzureRMDevTestLabSchedule_autoShutdowntaskNameTaskTypeMismatch(ri, location)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMDevTestLabScheduleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      missingRecurrenceConfig,
+				ExpectError: regexp.MustCompile("MissingRequiredProperties"),
+			},
+			{
+				Config:      missingDailyRecurrenceConfig,
+				ExpectError: regexp.MustCompile("MissingRequiredProperty"),
+			},
+			{
+				Config:      incorrectTaskNameConfig,
+				ExpectError: regexp.MustCompile("IncorrectResourceName"),
+			},
+			{
+				Config:      taskNameTaskTypeMismatchConfig,
+				ExpectError: regexp.MustCompile("IncorrectResourceName"),
+			},
```

Comment:
since this is validating the Azure API responses rather than the Terraform resource, this'll be a brittle test (since these can change over time) and as such is something we can't really rely on - can we instead add this logic to the Resource itself and test that?

Reply:
Agree, at microsoft.devtestlab/2016-05-15/schedules none of these validations have been explicitly mentioned. I will remove the TestAccAzureRMDevTestLabSchedule_autoShutdownPropertyRulesCheck function.